doc/source/data/dataset-advanced.rst

```diff
@@ -0,0 +1,319 @@
+
+With our Pandas extension type, :class:`TensorDtype <ray.data.extensions.tensor_extension.TensorDtype>`, and extension array, :class:`TensorArray <ray.data.extensions.tensor_extension.TensorArray>`, you can do familiar aggregations and arithmetic, comparison, and logical operations on a DataFrame containing a tensor column and the operations will be applied to the underlying tensors as expected. With our Arrow extension type, :class:`ArrowTensorType <ray.data.extensions.tensor_extension.ArrowTensorType>`, and extension array, :class:`ArrowTensorArray <ray.data.extensions.tensor_extension.ArrowTensorArray>`, you'll be able to import that DataFrame into Ray Datasets and read/write the data from/to the Parquet format.
+
+Automatic conversion between the Pandas and Arrow extension types/arrays keeps the details under-the-hood, so you only have to worry about casting the column to a tensor column using our Pandas extension type when first ingesting the table into a ``Dataset``, whether from storage or in-memory. All table operations downstream from that cast should work automatically.
+
+Reading existing serialized tensor columns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you already have a Parquet dataset with columns containing serialized tensors, you can have these tensor columns cast to our tensor extension type at read-time by giving a simple schema for the tensor columns. Note that these tensors must have been serialized as their raw NumPy ndarray bytes in C-contiguous order (e.g. serialized via ``arr.tobytes()``).
+
+.. code-block:: python
+
+    import numpy as np
+    import pandas as pd
+    from ray.data.extensions import TensorDtype, TensorArray
+
+    path = "/tmp/some_path"
+
+    # Create a DataFrame with a list of serialized ndarrays as a column.
+    # Note that we do not cast it to a tensor array, so each element in the
+    # column is an opaque blob of bytes.
+    arr = np.arange(24).reshape((3, 2, 2, 2))
+    df = pd.DataFrame({
+        "one": [1, 2, 3],
+        "two": [tensor.tobytes() for tensor in arr]})
+
+    # Write the dataset to Parquet. The tensor column will be written as an
+    # array of opaque byte blobs.
+    ds = ray.data.from_pandas([ray.put(df)])
+    ds.write_parquet(path)
+
+    # Read the Parquet files into a new Dataset, with the serialized tensors
+    # automatically cast to our tensor column extension type.
+    ds = ray.data.read_parquet(
+        path, _tensor_column_schema={"two": (np.int, (2, 2, 2))})
```

Comment:
>>> x = ray.data.read_parquet("/tmp/some_path", _tensor_column_schema={"two": (np.int, (2, 2, 2))})
>>> x
Dataset(num_blocks=1, num_rows=3, schema={one: int64, two: extension<arrow.py_extension_type<ArrowTensorType>>})
>>> x.take()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/eric/Desktop/ray/python/ray/data/impl/arrow_block.py", line 64, in __repr__
    return str(self)
  File "/home/eric/Desktop/ray/python/ray/data/impl/arrow_block.py", line 61, in __str__
    return "ArrowRow({})".format(self.as_pydict())
  File "/home/eric/Desktop/ray/python/ray/data/impl/arrow_block.py", line 33, in as_pydict
    return {k: v[0] for k, v in self._row.to_pydict().items()}
  File "pyarrow/table.pxi", line 1762, in pyarrow.lib.Table.to_pydict
  File "pyarrow/table.pxi", line 464, in pyarrow.lib.ChunkedArray.to_pylist
  File "pyarrow/array.pxi", line 1193, in pyarrow.lib.Array.to_pylist
AttributeError: 'numpy.ndarray' object has no attribute 'as_py'

Reply:
Ah, didn't test repr, will fix that and add a test.